fuzz/fuzz_targets/hexf_parse.rs

```diff
@@ -0,0 +1,78 @@
+#![no_main]
+
+use std::{ffi::CString, ptr};
+
+use errno::{errno, set_errno, Errno};
+use libc::strtod;
+use libfuzzer_sys::{arbitrary::Arbitrary, fuzz_target};
+
+use hexf_parse::{parse_hexf32, parse_hexf64};
+
+#[derive(Debug, Arbitrary)]
+enum TargetType {
+    F32,
+    F64,
+}
+
+// IMPORTANT: For the verification with strtof/d to work properly, your locale (LC_ALL) should be set C.UTF-8.
+//            Otherwise strtof/d will use your locale's decimal separator.
+
+macro_rules! verify_with_strto_fn {
+    ($target_name:ident, $strto_fn:ident, $string_to_parse:ident, $result:ident) => {
+        if $string_to_parse.contains("p") || $string_to_parse.contains("P") {
+            if let Ok(c_string) = CString::new($string_to_parse.replace('_', "")) {
+                unsafe {
+                    // strtod will write a *mut c_char to the character past the last interpreted character in here
+                    let mut end_ptr = ptr::null_mut();
+
+                    let strto_fn_result = $strto_fn(c_string.as_ptr(), &mut end_ptr);
+
+                    // See https://en.cppreference.com/w/c/string/byte/strtof
+                    // for a description of the behavior of strtof/d functions
+
+                    let strto_fn_errno = errno();
+                    if strto_fn_errno.0 != 0 {
+                        assert!(
+                            $result.is_err(),
+                            concat!(stringify!($target_name), " succeeded, but strtod failed with `{}`"),
+                            strto_fn_errno
+                        );
+                        set_errno(Errno(0))
+                    } else if *end_ptr != 0 {
```

Comment:
Should this be `!end_ptr.is_null()`?

Reply:
No, this should check whether the `end_ptr` points to the null byte at the end of the `CString`. Thanks for bringing it up though, since I should probably add a comment for that.

Reply:
I initially wanted to use `b'\0'`, but `c_char` is apparently `i8` (*shakes fist* C signed chars). Feel free to suggest something syntactically nicer if anything comes to mind.

Reply:
I don't mind `(b'\0' as c_char)`